x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java

```diff
@@ -331,10 +331,9 @@ private boolean checkIndexMappingUpToDate(ClusterState clusterState) {
         final SystemIndexDescriptor descriptor = systemIndexDescriptor.getDescriptorCompatibleWith(
             getMinSecurityIndexMappingVersion(clusterState)
         );
-        if (descriptor == null) {
-            return false;
-        }
-        return descriptor.getMappingsVersion().version() <= loadIndexMappingVersion(systemIndexDescriptor.getAliasName(), clusterState);
+
+        return descriptor == null
```

Comment:
This will return true if there is no descriptor that's compatible with the min version in the cluster.